pkg/util/iptables/iptables.go

```diff
@@ -203,11 +203,10 @@ type runner struct {
 
 // newInternal returns a new Interface which will exec iptables, and allows the
 // caller to change the iptables-restore lockfile path
-func newInternal(exec utilexec.Interface, protocol Protocol, lockfilePath string) Interface {
+func newInternal(exec utilexec.Interface, protocol Protocol, lockfilePath string) (Interface, error) {
 	version, err := getIPTablesVersion(exec, protocol)
 	if err != nil {
-		klog.Warningf("Error checking iptables version, assuming version at least %s: %v", MinCheckVersion, err)
-		version = MinCheckVersion
+		return nil, err
```

Comment:
should we differentiate between an error parsing the output from another error?
ie. return only if ExitStatus() != 0 and warn if we are not able to parse the version?

Reply:
We shouldn't ever have an error parsing the output though. If we can't parse the output, the most likely reason is that we aren't actually talking to an iptables binary...

I guess maybe they could change the format of "iptables --version" but that seems very unlikely